project-codeflare/codeflare-sdk
# Cli submit delete raycluster

# Issue link
Closes #239 

# What changes have been made
- Adds `submit raycluster` and `delete raycluster` to the CLI so the user can submit and delete to and from their Kubernetes cluster from the terminal. Unit tests included.
- Changed CLI functionality slightly so authentication, namespace, and cli-generated `.codeflare` folder are loaded and saved for each command to access.
- Added `Cluster.from_definition_yaml` command to get a `Cluster` object from a defined yaml file.

# Verification steps
After building SDK, you can run `codeflare` to see a list of all commands. 
- To test submit command, make sure you define a raycluster using `codeflare define raycluster` and then you can submit it using `codeflare submit raycluster` specifying name
- To test delete command, run `codeflare delete raycluster` specifying name and namespace.
You can also run the unit tests with `python -m pytest tests/unit_test.py` or `pytest -v tests/unit_test.py`.

## Checks
- Testing Strategy
